platform/wm/rhodes/browser/CEBrowserEngine.cpp

```diff
@@ -1157,4 +1164,134 @@ BOOL CEBrowserEngine::ZoomTextOnTab(int nZoom, UINT iTab)
 }
 
 
+DWORD WINAPI CEBrowserEngine::NetworkWindowThread( LPVOID lpParameter )
+{
+
+	LOG(INFO) + "CEBrowserEngine::NetworkWindowThread Started";
+	CEBrowserEngine * pEng = (CEBrowserEngine*) lpParameter;
+
+	TCHAR szWindowText[MAX_PATH];
+	TCHAR szWindowClass[MAX_PATH];
+	HWND hCurrentWindow = NULL;
+
+	ZeroMemory(szWindowText,MAX_PATH);
+	ZeroMemory(szWindowClass,MAX_PATH);
+
+
+	LOG(INFO) + "CEBrowserEngine::NetworkWindowThread";
+	LOG(INFO) + pEng->m_bNavigationComplete;
+	
+	while(!pEng->m_bNavigationComplete)
+	{
+		HWND hCurrentWindow=FindWindow(TEXT("Dialog"),TEXT("Enter Network Password"));
+		if(hCurrentWindow != NULL)
+		{
+
+#ifdef _DEBUG
+			GetWindowText(hCurrentWindow,szWindowText,MAX_PATH);
+			LOG(INFO) + "CEBrowserEngine::NetworkWindowThread FindWindow  Title";
+			LOG(INFO) + szWindowText;
+			ZeroMemory(szWindowClass,MAX_PATH);
+			GetClassName(hCurrentWindow,szWindowClass,MAX_PATH);
+			LOG(INFO) + "CEBrowserEngine::NetworkWindowThread FindWindow  class";
+			LOG(INFO) + szWindowClass;
+			LONG lIndex = GetWindowLong(hCurrentWindow,GWL_ID);
+			LOG(INFO) + "CEBrowserEngine::NetworkWindowThread FindWindow  id";
+			LOG(INFO) + lIndex;
+#endif
+			::EnumChildWindows(hCurrentWindow,(WNDENUMPROC)NetworkWndProc,0);
+			LOG(INFO) + "CEBrowserEngine::NetworkWindowThread Ended (0)";
+			pEng->m_bNavigationComplete = TRUE;
+			return 0;	
+		}
+		
+	}
+
+	LOG(INFO) + "CEBrowserEngine::NetworkWindowThread Ended";
+	pEng->m_bNavigationComplete = TRUE;
```

Comment:
This variable is false at onbeforenavigate.
If window is found then it set to true and return in while loop itself.
Another scenario is if windows is not and navigation is complete then it will be set to true for proper thread exit.
________________________________
From: Abhineet Agarwal<mailto:notifications@github.com>
Sent: ý15-ý01-ý2015 22:44
To: rhomobile/rhodes<mailto:rhodes@noreply.github.com>
Cc: RAGHAVENDRA RAO, LAKSHMI NARASIMMAN<mailto:KQNG43@zebra.com>
Subject: Re: [rhodes] /EMBPD00158491 - [SAP-ITS][CE5/MK4000]-Sip is not shown while trying to enter text in fields of Authentication screen (#461)


In platform/wm/rhodes/browser/CEBrowserEngine.cpp<https://github.com/rhomobile/rhodes/pull/461#discussion-diff-23025793>:

> +                     LOG(INFO) + "CEBrowserEngine::NetworkWindowThread FindWindow  class";
> +                     LOG(INFO) + szWindowClass;
> +                     LONG lIndex = GetWindowLong(hCurrentWindow,GWL_ID);
> +                     LOG(INFO) + "CEBrowserEngine::NetworkWindowThread FindWindow  id";
> +                     LOG(INFO) + lIndex;
> +#endif
> +                     ::EnumChildWindows(hCurrentWindow,(WNDENUMPROC)NetworkWndProc,0);
> +                     LOG(INFO) + "CEBrowserEngine::NetworkWindowThread Ended (0)";
> +                     pEng->m_bNavigationComplete = TRUE;
> +                     return 0;
> +             }
> +
> +     }
> +
> +     LOG(INFO) + "CEBrowserEngine::NetworkWindowThread Ended";
> +     pEng->m_bNavigationComplete = TRUE;


What i mean to say is that why are you setting it to true. This variable will already be true... It will reach to that line only if it exits from top while loop.

—
Reply to this email directly or view it on GitHub<https://github.com/rhomobile/rhodes/pull/461/files#r23025793>.


________________________________
- CONFIDENTIAL-

This email and any files transmitted with it are confidential, and may also be legally privileged. If you are not the intended recipient, you may not review, use, copy, or distribute this message. If you receive this email in error, please notify the sender immediately by reply email and then delete this email.